### PR TITLE
Travis: add clean edge script

### LINKED ISSUE #790: edge release mixes new assets & old source code, commit, date
Currently, the edge release's notes get updated to the current master commit and its assets get updated by Travis. So far, so good.
However, the release's date and associated commit remain in their original state. The same applies to the source code assets (zip, tar.gz), which remain at the original commit.

That's not just less than ideal, but even confusing and plain incorrect. I'd prefer to re-create the GitHub release on each update of the edge tag. Ping @vespakoen.

https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#delete-a-release

### PR DESCRIPTION
**Pull request by vespakoen**
This fixes #790 

Thanks @ppd for providing the code!

### CODE REVIEW

**Comment by vespakoen** [comment]
I also want to add the script to remove the last tag as well, will try to do so tomorrow.

File: .travis/remove-edge.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+old_release=$(curl \
+  -H "Accept: application/vnd.github.v3+json" \
+  https://api.github.com/repos/vespakoen/solvespace/releases/tags/edge \
+  | jq -r ".url")
+
+curl \
+  -X DELETE \
+  -H "Accept: application/vnd.github.v3+json" \
+  -H "Authorization: token $GITHUB_TOKEN" \
+  "$old_release"
```
Comment (ppd) [inline .travis/remove-edge.sh:15]: nit: Newline at end of file

File: .travis/remove-edge.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+sudo apt-get update -qq
+sudo apt-get install jq
```
Comment (ppd) [inline .travis/remove-edge.sh:4]:
```suggestion
sudo apt-get -y install jq
```

File: .travis/remove-edge.sh [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
```
Comment (ppd) [inline .travis/remove-edge.sh:1]:
```suggestion
#!/bin/sh -xe
```

**Comment by phkahler** [comment]
@vespakoen ??? 

I know how it goes, I've been busy all week. :-)

**Comment by vespakoen** [comment]
Hey! It isn't a quick fix like I thought it would be, I will look into it again in a bit.

When I was trying this out travis had a huge backlog of builds and it was taking forever too see if it was working or not, then life happened indeed ;)

Hopefully I'll have more luck now & thanks for the suggestions ppd!

**Comment by vespakoen** [comment]
Damnit, I just force pushed the change to the solvespace repository, had my remote setup incorrectly...
Do any of you guys have the correct latest version? otherwise i'll have to get them from the PR's again...

**Comment by vespakoen** [comment]
@phkahler It looks like you did the last push on the master branch before I screwed up (sha: 54c57286e62783f347214618b8b35c99dd3936ca, message: Add end marker to text window and increase max rows.)

Can you force push that to the master branch again? I overwrote it with my own branch which was behind a couple of commits.

This PR is not working by the way and can be merged (once the master branch is fixed again)

Sorry for the trouble...

**Comment by phkahler** [comment]
No problem. So you want me to revert one commit?

**Comment by phkahler** [comment]
@vespakoen  Oh, I see. You nuked a couple commits. I think I still have those branches here.

**Comment by phkahler** [comment]
@vespakoen  I think you just lost one commit. I rebased master on my branch for issue779 and pushed it up. So your commit is still there but has one inserted before it. If you want that one removed I can go back one it (or you can). Or if your commit is good and it's all back then we can leave it?

**Comment by phkahler** [comment]
@vespakoen  Nope, looks like we lost #789 as well.

**Comment by vespakoen** [comment]
Hey, my fork was at 6ff8db93e83d06926d182c4d9d66874667ae2cb8 (Use zOrder for selections within the same group)

I think there were 2 commits after it that I nuked:
- Add end marker to text window and increase max rows
- Update README.md

The `Travis: add clean edge script` that is now on the master branch is an old version of this PR, it should also be removed / updated with the one from this PR.

I thought you still had the correct state of things and could force push it onto master again, but I guess that is not the case.

I will sort this out now...

**Comment by vespakoen** [comment]
@phkahler Looks like you are working on it, so I'll leave you to it, If you still have that Update README.md commit, you can include it in master again, otherwise, here is a way to get the diff:

https://github.com/solvespace/solvespace/commit/f392189aa003d74968bc7876c805b041645abb4a.diff

**Comment by phkahler** [comment]
@vespakoen I'm having some trouble. My local master now shows the old version of the readme but the commit is not in my log. That means I'm going too fast and need to pause. Feel free to try to fix it ;-)

As far as I can tell the log should end with:

Update the readme
Use zOrder for selections....
adjust logo size

Maybe those top two are is reverse order, but that's not really important? I never pulled master after merging the readme one, so I couldn't just push it back.

**Comment by vespakoen** [comment]
The order is all weird now for the last 3 commits but I think everything is on there again...

**Comment by ruevs** [comment]
I have up to the Z order in my fork
https://github.com/ruevs/solvespace
And I think everything but the readme in a local repo, but that will have to wait until tomorrow... If you want it.

**Comment by ppd** [comment]
I think we should configure `master` as a protected branch to prevent force pushing to it.

**Comment by vespakoen** [comment]
@ppd I agree! In Github it's also possible. I use it on another project.

https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/configuring-protected-branches

**Pull request merged**